mindsdb_server/namespaces/datasource.py

```diff
@@ -66,11 +67,108 @@ def save_datasource_metadata(ds):
         with open(os.path.join(mindsdb.CONFIG.MINDSDB_DATASOURCES_PATH, ds['name'], 'metadata.json'), 'w') as fp:
             json.dump(ds, fp)
 
-def create_sqlite_db(path, ds):
+def create_sqlite_db(path, data_frame):
     con = sqlite3.connect(path)
-    ds.to_sql(name='data', con=con, index=False)
+    data_frame.to_sql(name='data', con=con, index=False)
     con.close()
 
+def get_analysis(source):
+    global global_mdb
+    return global_mdb.analyse_dataset(source, sample_margin_of_error=0.025)
+
+def cast_df_columns_types(df):
+    types_map = {
+        'Numeric': {
+            'Int': 'int64',
+            'Float': 'float64',
+            'Binary': 'bool'
+        },
+        'Date': {
+            'Date': 'datetime64',       # YYYY-MM-DD
+            'Timestamp': 'datetime64'   # YYYY-MM-DD hh:mm:ss or 1852362464
+        },
+        'Categorical': {
+            'Binary Category': 'category',
+            'Category': 'category'
+        },
+        'File Path': {
+            'Image': 'object',
+            'Video': 'object',
+            'Audio': 'object'
+        },
+        'Url': {},
+        'Sequential': {
+            'Text': 'object',
+            'Array': 'object'
+        }
+    }
+
+    analysis = get_analysis(df)
+    columns = [dict(name=x) for x in list(df.keys())]
+
+    for column in columns:
+        try:
+            name = column['name']
+            col_type = analysis['data_analysis_v2'][name]['typing']['data_type']
+            col_subtype = analysis['data_analysis_v2'][name]['typing']['data_subtype']
+            new_type = types_map[col_type][col_subtype]
+            if new_type == 'int64' or new_type == 'float64':
+                df[name] = df[name].apply(lambda x: x.replace(',','.'))
+            if new_type == 'int64':
+                df = df.astype({name: 'float64'})
```

Comment:
Why are we seemingly making float64 columns out of int64 columns?

Reply:
For int64 column is two casts here: firts is cast to 'float64', second to new_type (int64). That because some values in columns analyzed as int-type has values with dot. If do just df.astype({name: 'int64'}) on column with string value, for example '10.1' - it will raise error.